jupyterhub/spawner.py

```diff
@@ -81,6 +81,7 @@ class Spawner(LoggingConfigurable):
         'VIRTUAL_ENV',
         'LANG',
         'LC_ALL',
+        'SHELL'
```

Comment:
The SHELL of the server terminal (often sh) user shouldn't set the SHELL of the single-user servers.